mlflow/utils/rest_utils.py

```diff
@@ -77,6 +89,37 @@ def http_request(
     Returns:
         requests.Response object.
     """
+    if host_creds.databricks_workspace_client:
+        try:
+            if method == "GET":
+                query = kwargs.get("params") or kwargs.get("json")
+                extra_kwargs = {"query": query}
+            else:
+                extra_kwargs = {"body": kwargs.get("json")}
+
+            raw_response = host_creds.databricks_workspace_client.api_client.do(
+                method=method,
+                path=endpoint,
+                headers=extra_headers,
+                raw=True,
+                **extra_kwargs,
+            )
+            return raw_response["contents"]._response
+        except Exception as e:
+            if hasattr(e, "error_code"):
```

Comment:
Does `api_client.do` throw `HTTPError`?

Reply:
No. it throws errors defined in https://github.com/databricks/databricks-sdk-py/blob/main/databricks/sdk/errors/platform.py

Reply:
Why not catch `DatabricksError` instead?

Reply:
https://github.com/databricks/databricks-sdk-py/blob/db1f4ae9d055ff62f79374cbdfedf0e1da87cdf6/databricks/sdk/core.py#L167

- `self._make_nicer_error` performs error conversion.
- No `self._make_nicer_error` usage in `do`.

Reply:
> Why not catch DatabricksError instead?

Sounds good.

Reply:
I didn't check this part code, but we can do a simple test to know to raises `DatabricksError`:

```
os.environ["DATABRICKS_HOST"] = ...
os.environ["DATABRICKS_TOKEN"] = ...
mlflow.set_tracking_uri("databricks")
mlflow.create_experiment("/Users/weichen.xu@databricks.com/my-exp-002")
mlflow.create_experiment("/Users/weichen.xu@databricks.com/my-exp-002") # raise 
```

Reply:
let's test it. The code says do doesn't throw these errors.

Reply:
looks like retryable throws.